ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala

```diff
@@ -3,27 +3,27 @@
 
 package com.daml.platform.store
 
+import com.daml.ledger.resources.{ResourceContext, ResourceOwner}
 import com.daml.logging.{ContextualizedLogger, LoggingContext}
 import com.daml.platform.configuration.ServerRole
 import com.daml.platform.store.FlywayMigrations._
 import com.daml.platform.store.dao.HikariConnection
-import com.daml.resources.ResourceOwner
 import com.zaxxer.hikari.HikariDataSource
 import org.flywaydb.core.Flyway
 import org.flywaydb.core.api.MigrationVersion
 import org.flywaydb.core.api.configuration.FluentConfiguration
 
+import scala.concurrent.Future
 import scala.concurrent.duration.DurationInt
-import scala.concurrent.{ExecutionContext, Future}
 
 private[platform] class FlywayMigrations(jdbcUrl: String)(implicit loggingContext: LoggingContext) {
   private val logger = ContextualizedLogger.get(this.getClass)
 
   private val dbType = DbType.jdbcType(jdbcUrl)
 
-  def validate()(implicit executionContext: ExecutionContext): Future[Unit] =
+  def validate()(implicit resourceContext: ResourceContext): Future[Unit] =
     dataSource.use { ds =>
-      Future {
+      Future.successful {
```

Comment:
Can you double check that this change is ok? I believe it's ok because I guess `dataSource.use` wraps the `Future` so that the `Throwable` is caught correctly, but it's difficult to say from GitHub. Same for the two following analogous changes.

Reply:
It is, because `use` calls the behavior inside a `flatMap`. There's a unit test for it.